teps/0044-decouple-task-composition-from-scheduling.md

```diff
@@ -84,97 +84,399 @@ that can be shared between them, and running the second Task will be delayed by
-* [Introduce scheduling rules to Pipeline](#introduce-scheduling-rules-to-pipeline)
-* [PipelineRun: emptyDir](#pipelinerun-emptydir)
-* [Controller configuration](#controller-level)
-* [Within the Task](#within-the-task)
-* [Remove distinction between Tasks and Pipelines](#remove-distinction-between-tasks-and-pipelines)
-* [Custom Pipeline](#custom-pipeline)
-* [Create a new Grouping CRD](#create-a-new-grouping-crd)
-* [Rely on the Affinity Assistant](#rely-on-the-affinity-assistant)
-* [Custom scheduler](#custom-scheduler)
-* [Support other ways to share data (e.g. buckets)](#support-other-ways-to-share-data-eg-buckets)
-* [Focus on workspaces](#focus-on-workspaces)
-
-Most of the solutions above involve allowing more than one Task to be run in the same pod, and those proposals all share
-the following pros & cons.
+[Implementation options](#implementation-options):
+  * In the general theme of running [more than one Task in one pod](#more-than-one-task-in-one-pod):
+    * [Pipeline executed as pod](#pipeline-executed-as-pod)
+    * [Pipeline executed as TaskRun](#pipeline-executed-as-taskrun)
+    * [TaskRun controller composes pods from multiple TaskRuns](#taskrun-controller-composes-one-pod-from-multiple-taskruns)
+* [Other implementation options](#other-implementation-options):
+  * [Rely on the Affinity Assistant](#rely-on-the-affinity-assistant)
+  * [Custom scheduler](#custom-scheduler)
+  * [Support other ways to share data (e.g. buckets)](#support-other-ways-to-share-data-eg-buckets)
+
+[Syntax options](#syntax-options) for running more than one Task in one pod:
+* [At authoring time](#at-authoring-time)
+  * [In the Pipeline](#in-the-pipeline)
+    * [Task composition in Pipeline Tasks](#task-composition-in-pipeline-tasks)
+      * [Using pipelines in pipelines](#-using-pipelines-in-pipelines)
+      * [init/before finally/after](#-initbefore-finallyafter)
+    * [Automatically combine Tasks based on workspace use](#automagically-combine-tasks-based-on-workspace-use)
+    * [Introduce scheduling rules to Pipeline](#introduce-scheduling-rules-to-pipeline)
+    * [Focus on workspaces](#focus-on-workspaces)
+  * [In the Task](#within-the-task)
+  * [Other](#other-authoring-time-options)
+    * [Remove distinction between Tasks and Pipelines](#remove-distinction-between-tasks-and-pipelines)
+    * [Create a new Grouping CRD](#create-a-new-grouping-crd)
+    * [Custom Pipeline](#custom-pipeline)
+    * [Update PipelineResources to use Tasks](#update-pipelineresources-to-use-tasks)
+* [At runtime](#runtime-instead-of-authoring-time)
+  * [PipelineRun: emptyDir](#pipelinerun-emptydir)
+  * [Controller configuration](#controller-level)
+
+### Implementation options
+
+This section describes various ways that the functionality we're proposing could be implemented, which we can make
+choices around somewhat independently of deciding
+[what syntax we want to create to support this](#syntax-options).
+
+#### More than one Task in one pod
```

Comment:
Nit, but why is it the only alternative at this level, and all others grouped under "other implementation options" ? 
This TEP is about "Decouple Task Composition from Scheduling", more than one task in one pod is just one option among other, I don't see it as the main possibility (yet?)